File: app/src/main/main/java/com/timeback/ui/app/MainActivity.java
package com.timeback.ui.app;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import dagger.hilt.android.AndroidEntryPoint;

/**
 * Single Activity — Navigation Component로 Fragment를 교체한다.
 * Jetpack Compose 대신 Fragment + XML 방식 사용 (Java 호환).
 */
@AndroidEntryPoint
public class MainActivity extends AppCompatActivity {

    private NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // TODO: setContentView(R.layout.activity_main);
        // NavHostFragment를 호스트로 사용
        // NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager()
        //         .findFragmentById(R.id.nav_host_fragment);
        // navController = navHostFragment.getNavController();
    }
}

File: src/main/java/com/timeback/ui/app/MainActivity.kt
package com.timeback.ui.app

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.ui.Modifier
import androidx.navigation.compose.rememberNavController
import dagger.hilt.android.AndroidEntryPoint

/**
 * TimeBack 단일 Activity
 * Hilt DI + Navigation Compose 진입점
 */
@AndroidEntryPoint
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                Surface(modifier = Modifier.fillMaxSize()) {
                    val navController = rememberNavController()
                    TimeBackNavGraph(navController = navController)
                }
            }
        }
    }
}
